tasks/pipeline.py

```diff
@@ -160,6 +161,37 @@ def trigger(
     )
 
 
+@task
+def auto_cancel_previous_pipelines(ctx):
+    """
+    Automatically cancel previous pipelines running on the same ref
+    """
+
+    project_name = "DataDog/datadog-agent"
+    if not os.environ.get('GITLAB_TOKEN'):
+        raise Exit("GITLAB_TOKEN variable needed to cancel pipelines on the same ref.", 1)
+
+    gitlab = Gitlab(project_name=project_name, api_token=get_gitlab_token())
+    gitlab.test_project_found()
+
+    git_ref = os.getenv("CI_COMMIT_REF_NAME")
+    git_sha = os.getenv("CI_COMMIT_SHA")
+
+    pipelines = get_running_pipelines_on_same_ref(gitlab, git_ref)
+    pipelines_without_current = [p for p in pipelines if p["sha"] != git_sha]
+
+    for pipeline in pipelines_without_current:
+        # We cancel pipeline only if it correspond to a commit that is an ancestor of the current commit
+        try:
+            ctx.run(f'git merge-base --is-ancestor {pipeline["sha"]} {git_sha}')
+            print(
+                f'Gracefully canceling jobs that are not canceled on pipeline {pipeline["id"]} ({pipeline["web_url"]})'
+            )
+            gracefully_cancel_pipeline(gitlab, pipeline, force_cancel_stages=["package_build"])
+        except Exception:
+            print(f'{pipeline["sha"]} is not an ancestor of {git_sha}')
```

Comment:
Could you use a specific Exception instead of the general Exception keyword ?
Also printing the error as well as printing you message might be good (in case we might debug it later). WDYT?

Reply:
Or maybe remove the `try` keyword. To explain my point a bit, for now the logic is if `git merge-base --is-ancestor commitA commitB` exit code is 1, then it means `commitA` is not an ancestor of `commitB`.

The problem is, let's say `git` changes and the command now becomes `git merge-base-blabla` instead. Then you'll have a `Command not found` error returning a 1 exit code. And we'll have printed `commitA is not an ancestor of commitB` hence it's not the case. Keeping it this way will silenced every other error we might encounter.